numpy/lib/_shape_base_impl.py

```diff
@@ -1260,17 +1276,32 @@ def tile(A, reps):
         tup = tuple(reps)
     except TypeError:
         tup = (reps,)
-    d = len(tup)
     if all(x == 1 for x in tup) and isinstance(A, _nx.ndarray):
         # Fixes the problem that the function does not make a copy if A is a
         # numpy array and the repetitions are 1 in all dimensions
-        return _nx.array(A, copy=True, subok=True, ndmin=d)
+        return _nx.array(A, copy=True, subok=True, ndmin=len(tup))
     else:
         # Note that no copy of zero-sized arrays is made. However since they
         # have no data there is no risk of an inadvertent overwrite.
-        c = _nx.array(A, copy=False, subok=True, ndmin=d)
-    if (d < c.ndim):
-        tup = (1,)*(c.ndim-d) + tup
+        c = _nx.array(A, copy=False, subok=True, ndmin=len(tup))
```

Comment:
One thing I somewhat dislike about this is that axis is now an index into this new array, whose dimensions may have changed if `len(tup) > A.ndim`. Maybe this should be `ndmin=(len(tup) if axis is None else 0)`.

(If we go for that, that needs a test too!)